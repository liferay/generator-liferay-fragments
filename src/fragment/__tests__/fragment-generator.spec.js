const fs = require('fs');
const path = require('path');
const YeomanTest = require('yeoman-test');

/**
 * @param  {string[]} paths
 */
function expectFile(...paths) {
  return expect(
    fs.readFileSync(path.join(...paths), 'utf-8')
  ).toMatchSnapshot();
}

/**
 * @param  {string} base
 * @param  {string[]} paths
 */
function expectFiles(base, paths) {
  return paths.map((_path) => expectFile(path.join(base, _path)));
}

describe('fragment-generator', () => {
  it('generates a new fragment', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withOptions({ minLiferayVersion: '7.2.0' })
      .withOptions({ fragmentName: 'Sample Fragment' })
      .withOptions({ fragmentType: 'section' })
      .withOptions({ fragmentCollectionSlug: 'sample-collection' })
      .then((projectPath) => {
        expectFiles(
          path.join(projectPath, 'src', 'sample-collection', 'sample-fragment'),
          ['fragment.json', 'index.html', 'styles.css', 'main.js']
        );
      }));

  it('needs a name', () =>
    new Promise((resolve, reject) => {
      YeomanTest.run(path.join(__dirname, '..'))
        .withOptions({ minLiferayVersion: '7.2.0' })
        .withOptions({ fragmentType: 'section' })
        .withOptions({ fragmentCollectionSlug: 'sample-collection' })
        .then(reject)
        .catch(resolve);
    }));

  it('needs a collection', () =>
    new Promise((resolve, reject) => {
      YeomanTest.run(path.join(__dirname, '..'))
        .withOptions({ minLiferayVersion: '7.2.0' })
        .withOptions({ fragmentName: 'Sample Fragment' })
        .withOptions({ fragmentType: 'section' })
        .then(reject)
        .catch(resolve);
    }));

  it('generates new features if supported', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withOptions({ minLiferayVersion: '7.3.3' })
      .withOptions({ fragmentName: 'Sample Fragment' })
      .withOptions({ fragmentType: 'section' })
      .withOptions({ fragmentCollectionSlug: 'sample-collection' })
      .then((projectPath) => {
        expectFiles(
          path.join(projectPath, 'src', 'sample-collection', 'sample-fragment'),
          ['fragment.json', 'index.html', 'styles.css', 'main.js']
        );
      }));
});
