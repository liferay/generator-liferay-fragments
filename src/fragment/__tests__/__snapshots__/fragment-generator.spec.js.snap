// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment-generator generates a new fragment 1`] = `
"{
  \\"cssPath\\": \\"styles.css\\",
  \\"configurationPath\\": \\"configuration.json\\",
  \\"htmlPath\\": \\"index.html\\",
  \\"jsPath\\": \\"main.js\\",
  \\"name\\": \\"Sample Fragment\\",
  \\"type\\": \\"section\\"
}
"
`;

exports[`fragment-generator generates a new fragment 2`] = `
"<div class=\\"sample-fragment\\">
  Sample Fragment
</div>
"
`;

exports[`fragment-generator generates a new fragment 3`] = `
".sample-fragment {
  color: inherit;
  font-size: 18px;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI',
    'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
    'Helvetica Neue', sans-serif;
}
"
`;

exports[`fragment-generator generates a new fragment 4`] = `
"console.group('Sample Fragment');
console.log('fragmentElement', fragmentElement);
console.groupEnd();
"
`;

exports[`fragment-generator generates new features if supported 1`] = `
"{
  \\"cssPath\\": \\"styles.css\\",
  \\"configurationPath\\": \\"configuration.json\\",
  \\"htmlPath\\": \\"index.html\\",
  \\"jsPath\\": \\"main.js\\",
  \\"name\\": \\"Sample Fragment\\",
  \\"type\\": \\"section\\"
}
"
`;

exports[`fragment-generator generates new features if supported 2`] = `
"<div class=\\"sample-fragment\\">
  <h1
    class=\\"[#if configuration.textColor.color?has_content]text-\${configuration.textColor.color}[/#if]\\"
    data-lfr-editable-id=\\"title\\"
    data-lfr-editable-type=\\"text\\"
  >
    Sample Fragment
  </h1>
</div>
"
`;

exports[`fragment-generator generates new features if supported 3`] = `
".sample-fragment {
  color: inherit;
  font-size: 18px;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI',
    'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
    'Helvetica Neue', sans-serif;
}
"
`;

exports[`fragment-generator generates new features if supported 4`] = `
"console.group('Sample Fragment');
console.log('fragmentElement', fragmentElement);
console.log('configuration', configuration);
console.groupEnd();
"
`;
